lib/irb/init.rb

```diff
@@ -395,42 +395,54 @@ def IRB.parse_opts(argv: ::ARGV)
   # Run the config file
   def IRB.run_config
     if @CONF[:RC]
-      rc_files.each do |rc|
-        # Because rc_file always returns `HOME/.irbrc` even if no rc file is present, we can't warn users about missing rc files.
-        # Otherwise, it'd be very noisy.
-        load rc if File.exist?(rc)
+      irbrc_files.each do |rc|
+        load rc
       rescue StandardError, ScriptError => e
         warn "Error loading RC file '#{rc}':\n#{e.full_message(highlight: false)}"
       end
     end
   end
 
   IRBRC_EXT = "rc"
-  def IRB.rc_file(ext = IRBRC_EXT)
-    warn "rc_file is deprecated, please use rc_files instead."
-    rc_files(ext).first
-  end
 
-  def IRB.rc_files(ext = IRBRC_EXT)
-    if !@CONF[:RC_NAME_GENERATOR]
-      @CONF[:RC_NAME_GENERATOR] ||= []
-      existing_rc_file_generators = []
+  def IRB.prepare_rc_name_generators
```

Comment:
Maybe we can declare this as private?
Also, I feel `prepare_irbrc_name_generators` is a clearer name.